lib/simp/beaker_helpers/inspec.rb

```diff
@@ -0,0 +1,169 @@
+module Simp::BeakerHelpers
+
+  # Helpers for working with Inspec
+  class Inspec
+    require 'highline'
+
+    # Create a new Inspec helper for the specified host against the specified profile
+    #
+    # @param sut
+    #   The SUT against which to run
+    #
+    # @param profile
+    #   The name of the profile against which to run
+    #
+    def initialize(sut, profile)
+      @sut = sut
+
+      @sut.install_package('inspec')
+
+      @sut_profile_dir = '/tmp/inspec_tests'
+
+      output_dir = File.absolute_path('sec_results/inspec')
+
+      unless File.directory?(output_dir)
+        FileUtils.mkdir_p(output_dir)
+      end
+
+      inspec_fixtures = File.join(fixtures_path, 'inspec_profiles')
+      os = fact_on(@sut, 'operatingsystem')
+      os_rel = fact_on(@sut, 'operatingsystemmajrelease')
+
+      @inspec_result_file = File.join(output_dir, "#{@sut.hostname}-inspec-#{Time.now.to_i}")
+
+      scp_to(@sut, File.join(inspec_fixtures, "#{os}-#{os_rel}-#{profile}"), @sut_profile_dir)
+
+      # The results of the inspec scan in Hash form
+      @inspec_results = {}
+    end
+
+    # Run the inspec tests and record the results
+    def run
+      sut_inspec_results = '/tmp/inspec_results.json'
+
+      inspec_cmd = "inspec exec --format json #{@sut_profile_dir} > #{sut_inspec_results}"
+      result = on(@sut, inspec_cmd, :accept_all_exit_codes => true)
+
+      tmpdir = Dir.mktmpdir
+      begin
+        Dir.chdir(tmpdir) do
+          scp_from(@sut, sut_inspec_results, '.')
+
+          local_inspec_results = File.basename(sut_inspec_results)
+
+          if File.exist?(local_inspec_results)
+            begin
+              @inspec_results = JSON.load(File.read(local_inspec_results))
+            rescue JSON::ParserError, JSON::GeneratorError
+              @inspec_results = nil
+            end
+          end
+        end
+      ensure
+        FileUtils.remove_entry_secure tmpdir
+      end
+
+      unless @inspec_results
+        File.open(@inspec_result_file + '.err', 'w') do |fh|
+          fh.puts(result.stderr.strip)
+        end
+
+        err_msg = ["Error running inspec command #{inspec_cmd}"]
+        err_msg << "Error captured in #{@inspec_result_file}" + '.err'
+
+        fail(err_msg.join("\n"))
+      end
+    end
+
+    # Output the report
+    #
+    # @param report
+    #   The inspec results Hash
+    #
+    def write_report(report)
```

Comment:
The answer to the rest of these will be understood when this gets pushed and the module can use it.